src/internal/util/argsArgArrayOrObject.ts

```diff
@@ -0,0 +1,32 @@
+/** @prettier */
+
+const { isArray } = Array;
+const { getPrototypeOf, prototype: objectProto, keys: getKeys } = Object;
+
+/**
+ * Used in functions where either a list of arguments, a single array of arguments, or a
+ * dictionary of arguments can be returned. Returns an object with an `args` property with
+ * the arguments in an array, if it is a dictionary, it will also return the `keys` in another
+ * property.
+ */
+export function argsArgArrayOrObject<T, O extends { [key: string]: T }>(args: T[] | [O] | [T[]]): { args: T[]; keys: string[] | null } {
```

Comment:
nitpick: I'd use `Record<string, T>` instead of `{ [key: string]: T }`. It's no big deal; it's just that I use the built-in types whenever possible.

Reply:
Normally I do too, TBH... but in this case, I feel like it communicated a little more about the use, I don't know. I guess `Record<string, T>` is more terse. I think I'll change it.